### ISSUE TITLE
Confusing error and no logging when initialising tokens

### ISSUE THREAD

**Issue by gijzelaerr** [opendnssec/SoftHSMv2#349]
OS X version `10.12.6`
SoftHSMv2 version `2.3.0`

I installed SoftHSMv2 following the instructions in `OSX-NOTES.md`. 

when I try to initialise the token storage I run into troubles:
```
$ softhsm2-util --init-token
ERROR: Could not initialize the PKCS#11 library/module: /usr/local/Cellar/softhsmv2/2.3.0/lib/softhsm/libsofthsm2.so
ERROR: Please check log files for additional information.
```
I tried to find some logging information, but it looks like nothing shows up in syslog. The actual problem is that the path the the token storage does not exist.

if i make the tokens folder it seems to work:
```
$ mkdir -p /usr/local/Cellar/softhsmv2/2.3.0/var/lib/softhsm/tokens/
$ softhsm2-util --init-token --free --label gijs
```

**Comment by bellgrim** [comment]
These are log messages but at the WARNING-level and the default configuration is set to ERROR.

Oct 26 14:03:07 comp softhsm2-util: ../../../../src/lib/object_store/ObjectStore.cpp(59): Failed to enumerate object store in /var/lib/softhsm/tokens2/
Oct 26 14:03:07 comp softhsm2-util: ../../../src/lib/SoftHSM.cpp(499): Could not load the object store

We could also change the following debug message to an error message, which would be the information you are looking for: 
https://github.com/opendnssec/SoftHSMv2/blob/develop/src/lib/object_store/Directory.cpp#L111

**Comment by bellgrim** [comment]
The "missing" log messages was because of #330 where the log level was changed. In systems which use p11-kit to register all available PKCS#11 modules, an unconfigured softhsm will always fail to load and print the log messages mentioned above.

So we can't fix both #330 and this issue by setting the log levels correct. The compromise was to implement so that softhsm2-util will check the configuration and report any issues before loading the PKCS#11 library. Thus no need to check the log messages.
